Cannot specify link libraries for target \"gencmp\" which is not built by\nthis project."

CMake runs into the following error:

```
[{
	"resource": "/home/z002m76a/Repos/gencmpclient/CMakeLists.txt",
	"owner": "cmake-configure-diags",
	"severity": 8,
	"message": "CMake Error at CMakeLists.txt:138 (target_link_libraries):Cannot specify link libraries for target \"gencmp\" which is not built by\nthis project.",
	"source": "CMake (target_link_libraries)",
	"startLineNumber": 138,
	"startColumn": 1,
	"endLineNumber": 138,
	"endColumn": 10000,
	"origin": "extHost3"
}]
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -135,7 +135,6 @@ else()
     set(COVERAGE_FLAGS --coverage -fprofile-arcs)
     add_compile_options(${COVERAGE_FLAGS})
     link_libraries(${COVERAGE_FLAGS})
-    target_link_libraries(${LIBGENCMP_NAME} ${COVERAGE_FLAGS})
   endif()
 endif()
 message(STATUS "build mode: ${CMAKE_BUILD_TYPE}")
@@ -160,6 +159,10 @@ add_library(${LIBGENCMP_NAME} SHARED
   "${PROJECT_SOURCE_DIR}/.github/workflows/build.yml"
   ${SRC_DIR}/genericCMPClient.c
 )
+
+if(UNIX AND NOT APPLE AND CMAKE_BUILD_TYPE MATCHES Debug)
+  target_link_libraries(${LIBGENCMP_NAME} ${COVERAGE_FLAGS})
+endif()
 
 add_executable(cmpClient
   ${SRC_DIR}/cmpClient.c

